fem/quadinterpolator.cpp

```diff
@@ -277,11 +276,17 @@ static void Eval2D(const int NE,
                   }
                }
             }
-            if (VDIM == 2 && (eval_flags & QI::DETERMINANTS))
+            if (eval_flags & QI::DETERMINANTS)
             {
-               // The check (VDIM == 2) should eliminate this block when VDIM is
-               // known at compile time and (VDIM != 2).
-               det(q,e) = kernels::Det<2>(D);
+               if (VDIM == 2) { det(q,e) = kernels::Det<2>(D); }
+               else
+               {
+                  DeviceTensor<2> j(D, 3, 2);
+                  const double E = j(0,0)*j(0,0) + j(1,0)*j(1,0) + j(2,0)*j(2,0);
+                  const double F = j(0,0)*j(0,1) + j(1,0)*j(1,1) + j(2,0)*j(2,1);
+                  const double G = j(0,1)*j(0,1) + j(1,1)*j(1,1) + j(2,1)*j(2,1);
+                  det(q,e) = sqrt(E*G - F*F);
+               }
```

Comment:
Sorry, I am still confused here. I see that we take the assumption that the vector dimension is equal to the spatial dimension. Where does this come from?

Reply:
These kernels are for computing the Jacobian determinants for the geometric factors. The "vdim" is the number of components of the nodal grid function, i.e. the spatial dimension.

If you have a 2D mesh, and a grid function with vdim 3, I believe the determinant calculation should work, even if the spatial dimension of the mesh isn't 3. But I don't see a particular use-case for that.

Reply:
Yeah seems like `j` is a `vdim x 2` `DeviceTensor`, and the `det(q,e)` calculation will be off for `vdim != 3`. 
Might be worth making sure that the highlighted code snippet is used only for the configuration it supports.

Reply:
Yes, this function is only ever called for dim = 2 and vdim = 3, [see here](https://github.com/mfem/mfem/pull/4089#discussion_r1478785346).

Reply:
> These kernels are for computing the Jacobian determinants for the geometric factors. The "vdim" is the number of components of the nodal grid function, i.e. the spatial dimension.
> 

Ah yes, sorry, my bad.

Should we leave some debug assert to check if vdim == 2 || vdim == 3 just in case, so if something goes wrong the error is clear, or will we get a clear message downstream in the function? I am thinking about adding back (and extending):

```
MFEM_VERIFY(VDIM == 2 || vdim == 3 || !(eval_flags & QI::DETERMINANTS), "");
```

Reply:
Done in 74e31a421. Does that look OK to you?